configure

```diff
@@ -1066,7 +1066,7 @@ echo "#define OCAML_STDLIB_DIR \"$libdir\"" >> s.h
 
 # Do #! scripts work?
 
-if (SHELL=/bin/sh; export SHELL; (./hashbang || ./hashbang2 || ./hashbang3) >/dev/null); then
+if ((./hashbang || ./hashbang2 || ./hashbang3 || printf "#!%s\nexit 1\n" `command -v cat` > hashbang; ./hashbang) >/dev/null); then
```

Comment:
This line seems a bit too long and too complex for its own good. Do I correctly understand that the new part is rewriting the file `config/auto-aux/hashbang`? Is this an intended side-effect? If you would like to build a new hashbang file programmatically (using `command -v` instead of assumptions on where `cat` is), maybe this could be done as a first step, on a new `hashbang4` file?